code/modules/mob/living/carbon/human/human_movement.dm

```diff
@@ -4,6 +4,12 @@
 	. += ..()
 	. += config.human_delay
 
+/mob/living/carbon/human/update_canmove()
+	if(!get_num_legs(1)) //get a number of USABLE legs (ignores limbs w/ AUGMENTABLE flag)
+		weakened += 1 //Kind of a hack but w/e
```

Comment:
... What?
Do you mean the slowdown stacking? If so, I'm not sure how bad is it. Needs testing.